### PR TITLE
Add a script to check for various inclusion criterias

### PR DESCRIPTION
**Pull request by adriantombu**
As promised [almost 3 months ago](https://github.com/rust-lang/arewewebyet/pull/437#issuecomment-2249868505), here's a script that checks for the following criterias for the existing crates listed on the repo:
- The crate has at least 4k recent downloads on crates.io
- The package's repository is not archived

I also added back crates I previously removed by error and cleaned existing ones that did not meet the above criterias.

Please note that I am by no means any good in bash scripting so I did my very best here and probably spent too much time writing those lines.

As a future addition in another PR, we can add the check to the last criteria (`"The package is not flagged as unmaintained in the Rust security advisory database"`) to clean some crates that have gone unmaintained for many years.

### CODE REVIEW

**Comment by LukeMathWalker** [comment]
Thanks for taking the time! 
Could we run it in CI, perhaps skipping the GitLab part, if it indeed requires a token?

File: scripts/check_criterias.sh [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      echo "** unable to retrieve api github data from ${REPOSITORY}"
+    fi
+
+    ARCHIVED=$(jq -r '.archived' <<< "$DATA")
+  elif [[ "$REPOSITORY" == *"bitbucket.org"* ]]; then
+    #DATA=$(curl -s -H "Accept:application/json" "https://api.bitbucket.org/2.0/repositories/$OWNER_REPO")
+    # No "archived" status on Bitbucket: https://jira.atlassian.com/browse/BCLOUD-18018
+    ARCHIVED="false"
+  elif [[ "$REPOSITORY" == *"gitlab.com"* ]]; then
+    DATA=$(curl -s -H "Accept:application/json" -H "Authorization: Bearer $GITLAB_TOKEN" "https://gitlab.com/api/v4/projects/$ENCODED_OWNER_REPO")
```
Comment (LukeMathWalker) [inline scripts/check_criterias.sh:75]: It doesn't look like we need auth for this? https://docs.gitlab.com/ee/api/projects.html#get-a-single-project

Comment (adriantombu) [reply scripts/check_criterias.sh:75]: Unfortunately an authenticated call to the Gitlab API only returns a fraction of the project fields and `archived` is not part of it.

It's not well described in their doc, you juste have a brief explanation of it in https://docs.gitlab.com/ee/api/projects.html#list-all-projects where they talk about

> When simple=true or the user is unauthenticated this returns something like:

File: scripts/check_criterias.sh [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    fi
+  done
+
+  #
+  # The package's repository is not archived
+  OWNER_REPO="${REPOSITORY#https://github.com/}"
+  ENCODED_OWNER_REPO="${OWNER_REPO//\//%2F}"
+
+  if [[ "$REPOSITORY" == *"github.com"* ]]; then
+    DATA=$(curl -s -H "Accept:application/json" -H "Authorization: Bearer $GITHUB_TOKEN" "https://api.github.com/repos/$OWNER_REPO")
```
Comment (LukeMathWalker) [inline scripts/check_criterias.sh:63]: Same here, we don't need a token for public repositories: https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#get-a-repository

Comment (adriantombu) [reply scripts/check_criterias.sh:63]: Unfortunately we need it, there's a rate limit of 60 api calls for unauthenticated users https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#primary-rate-limit-for-unauthenticated-users

**Pull request closed without merging**